### ISSUE TITLE
Slow cache?

### ISSUE THREAD

**Issue by canfiax** [Intervention/imagecache#38]
I just installed the library. Hitting the route:

{route}/{template}/{filename}

Is extremely slow. It takes 500ms or so, compared to direct image source which is 20ms or so.

I wonder if the image is even cached. I suspect not. Using memcached as driver on a Laravel 5 installation.. How to debug further?

**Comment by canfiax** [comment]
On a page with 10-20 images, they take 10-15 seconds to load, and some of them even fail, like this:

![image](https://cloud.githubusercontent.com/assets/3786627/6428374/71001aec-bfa0-11e4-9069-68441f5f3b3b.png)

**Comment by canfiax** [comment]
I've made a comparison:

Using imagecache:
![image](https://cloud.githubusercontent.com/assets/3786627/6428390/de9b3c9e-bfa0-11e4-8cc0-ae2e4c9160f4.png)

Using direct image source:
![image](https://cloud.githubusercontent.com/assets/3786627/6428395/eeb11658-bfa0-11e4-8aba-3612d99a0d13.png)

**Comment by olivervogel** [comment]
I suggest to create a XDebug profile, to discover what is causing this delay.

**Comment by ekeros** [comment]
canflax. Have the same problem, its faster to load image full size then cache'em. Please reply here if you find an answer! same for me

**Comment by DoubleTime** [comment]
Facing same issue,  almost extra 60-70ms in time latancy.

This is the code how I load the image at Route.php

    Route::get('images/cars/{src}', function ($src){    
        $cacheimage = Image::cache(function($image) use($src){
            return $image->make("images/products/".$src);
        },1440);

        return Response::make($cacheimage,200, array('Content-Type'=>'image/jpg'));
    });
In blade

````<img src="{{ URL::asset('/images/cars/theimage.jpg' }}" alt="">````

**Comment by mehrdd** [comment]
+1
is there any simple todo for real cache ?
i mean create a real thumbnail with Intervention's library.

**Comment by tshafer** [comment]
+1 I am also facing this issue. I am using it this way. http://image.intervention.io/use/url

**Comment by olivervogel** [comment]
I'm again suggesting to create a XDebug profile, to discover what exactly is causing this delay.

**Comment by tshafer** [comment]
Sorry about that.

Ill give it a try

**Comment by Elegant-Division** [comment]
Yep, getting pics from cache are so slow! Slow like first resizing.
http://c2n.me/3gUFhhr.jpg

**Comment by Elegant-Division** [comment]
NOTE!
Image from cache have always **status code 200**,
in normal situation images must have code **status code 304**.

I hope olivervogel will fix this!)

**Comment by jhellingman** [comment]
+1 Same issue with similar code.

**Comment by phyblack** [comment]
People, any thoughts on it?

**Comment by Nikita240** [comment]
+1 same issue, even though it does create cached image files in the storage, and the `Image::cache` function is returning a `CachedImage` object.

**Comment by jhellingman** [comment]
The issue is that the entire framework needs to be loaded for each image being retrieved, and that action is actually slower than most image manipulations. Resolved by generating direct urls to generated images, and let the server pull them directly from the cache, without use of the framework. Since the types of manipulations I want are very limited, I can keep the URLs to them in my database and write them out to generated pages. My conclusion: I have no need for this framework.

**Comment by aske-cph** [comment]
+1 - Seriously, it's useless as it is now.

Loading 8 images is very slow for me on a regular digital ocean droplet
It can sometimes take over 4 seconds to load 8 images at time. 

Sometimes the server hangs and loads all of the images at the same time. And i can see that the CPU spikes like crazy on the server, even though the images are cached.

I am going to just create a command i laravel that creates the thumbs with intervention, and delegates the generated thumbs and different sizes to appropriate folders instead of relying on this for now...

**Comment by thanhmabo** [comment]
I also has same problem, see my cache image, actually slowly http://www.thanhmabo.com/post

**Comment by cmnstmntmn** [comment]
any update on this ?

**Comment by wi-development** [comment]
hello?

**Comment by sunnyjayjay** [comment]
I agree. The caching is very much slower than loading the images directly. Is this problem going to be resolved or do we need to find another image caching solution?

**Comment by stephan-v** [comment]
Still slow as fuck, don't use this package.

**Comment by hijaumuda** [comment]
it's been 4 years, and still slow 👎

**Comment by zefexdeveloper** [comment]
Have anyone found a solution or an alternative for this?
